Tools/uorb_graph/create.py

```diff
@@ -68,189 +70,216 @@ def __init__(self, is_publication, topic_blacklist, orb_pub_sub_regexes, special
+            # just the matches for this particular pattern:
+            match = regex.search(source_line)
+
+            if match is None:
+                continue
+
+            # # all regexes should contain 3 capture groups  (or else this code block crashes)
+            # total_match = match.group(0)
+            route_group, topic_group = match.groups()
 
-        orb_id = 'ORB_ID('
+            log.debug(f"          ####:{self._name}:  {route_group}, {topic_group}")
 
-        for _, match in orb_pubsub_matches:
-            if match == 'ORB_ID_VEHICLE_ATTITUDE_CONTROLS': # special case
-                match = orb_id+orb_id_vehicle_attitude_controls_topic
+            # # TODO: handle this case... but not sure where, yet
+            # if match == 'ORB_ID_VEHICLE_ATTITUDE_CONTROLS': # special case
+            #     match = orb_id+orb_id_vehicle_attitude_controls_topic
 
             # match has the form: '[ORB_ID(]<topic_name>'
-            if match.startswith(orb_id):
-                topic_name = match[len(orb_id):]
-                self._add_topic(topic_name, file_name, module)
+            if route_group:
+                if route_group == 'ORB_ID':
+                    log.debug("            >>> Found ORB_ID topic: " + topic_group +  "    w/regex: " + str(regex.pattern))
+                    self._add_topic(matches, topic_group)
+                    break
+                elif route_group == '<' and topic_group.endswith('_s'):
+                    topic_group = topic_group[:-2]
+                    log.debug("            >>> Found C++ template-declaration: " + topic_group +  "    w/regex: " + str(regex.pattern))
+                    self._add_topic(matches, topic_group)
+                    # continue processing
+                elif route_group in ['{','('] and topic_group.endswith('_s'):
+                    topic_group = topic_group[:-2]
+                    log.debug("            >>> Found standard declaration: " + topic_group +  "    w/regex: " + str(regex.pattern))
+                    self._add_topic(matches, topic_group)
+                    break
+                elif route_group == '[':
+                    if topic_group.endswith('_s'):
+                        topic_group = topic_group[:-2]
+                        log.debug("            >>> Found array declaration: " + topic_group +  "    w/regex: " + str(regex.pattern))
+                        self._add_topic(matches, topic_group)
+                        break
+                    else:
+                        # no topic found -- ambiguity -- return an empty set
+                        return set()
+                elif 'Multi' in route_group and topic_group.endswith('_s'):
+                        topic_group = topic_group[:-2]
+                        log.debug("            >>> Found 'multi' declaration: " + topic_group +  "    w/regex: " + str(regex.pattern))
+                        self._add_topic(matches, topic_group)
+                        break
```

Comment:
If I see correctly, all of these match the data type, whereas it should match the message ID (topic instance)

Reply:
Fixing these issues at the regex definitions.

Reply:
(Although the '<' clause obviously gets removed)

Reply:
reworking these topics to match the new regexes.